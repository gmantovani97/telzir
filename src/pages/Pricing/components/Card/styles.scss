@import '../../../../styles/mixins.scss';
@import '../../../../styles/variables.scss';

@keyframes setShadow {
  from {
    box-shadow: 0 0 transparent;
  } to {
    box-shadow: 0 0 0.625rem $grey;
  }
}

.card {
  padding: 1.25rem;
  margin: 0 2.5rem;

  &--best {
    padding: 1.25rem;
    border: 0.125rem solid $primary-dark;
  }

  &:hover {
    animation: setShadow 500ms forwards;
  }

}

.flag {
  align-self: center;
  padding: 0.3125rem;
  height: 1.625rem;
  border-radius: 0.8125rem;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  background-color: $primary-color;

  &__icon {
    height: 1.25rem;
    fill: $white;
  }

  &__text {
    font-weight: 600;
    color: $white;
  }
}


.image {
  height: 4rem;
}

.title {
  align-self: center;
  margin: 1.25rem 0;
  letter-spacing: 0.0625rem;
  font-weight: 500;
  color: $secondary-dark;
}

.section {
  margin: 1.25rem 0;
  flex: 1;
  justify-content: center;

  &__item {
    flex-direction: row;
    margin: 0.3125rem 0;
  }

  &__icon {
    width: 1.25rem;
    margin-right: 0.3125rem;
    &--exclude {
      fill: #A00;;
    }
    &--include {
      fill: #0A0;
    }
  }

  &__text {
    font-size: 0.875rem;
  }

}

.price {
  justify-content: center;
  align-items: center;

  &__section {
    flex-direction: row;
    justify-content: center;
    align-items: center;
  }

  &__number {
    font-size: 1.25rem;
    margin-right: 0.125rem;
  }


  &__text {
    font-size: 0.875rem;
  }

}

@keyframes setBestButtonColor {
  from {
    background-color: $secondary-color;
    border: 0.125rem solid $secondary-color;
  } to {
    background-color: $secondary-dark;
    border: 0.125rem solid $secondary-dark;
  }
}

@keyframes setButtonColor {
  from {
    background-color: $white;
    border: 0.125rem solid $secondary-dark;
    p {
      color: $secondary-dark;
    }
  } to {
    background-color: $secondary-dark;
    border: 0.125rem solid $secondary-dark;
    p {
      color: $white;
    }
  }
}

@keyframes setButtonTextColor {
  from {
    color: $secondary-dark;
  } to {
    color: $white;
  }
}

@keyframes setBestButtonTextColor {
  to {
    color: $white;
  }
}



.button {
  background-color: $white;
  height: 3rem;
  justify-content: center;
  align-items: center;
  border-radius: 1.5rem;
  margin-top: 1.25rem;
  cursor: pointer;
  border: 0.125rem solid $secondary-dark;

  &:hover {
    animation: setButtonColor 500ms forwards;

    p {
      color: $white;
    }
  }

  &--best {
    background-color: $secondary-color;
    border: 0.125rem solid $secondary-color;

    &:hover {
      animation: setBestButtonColor 500ms forwards;
    }
  
  }

  &__text {
    color: $secondary-dark;
    font-weight: 800;
    
    &--best {
      color: $white;
    }
  }
}
