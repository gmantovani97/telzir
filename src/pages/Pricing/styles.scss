@import '../../styles/mixins.scss';
@import '../../styles/variables.scss';

.container {
  flex: 1;
  padding: 1.25rem;

  &__title {
    align-self: center;
    font-weight: 300;
    text-align: center;
    @include s {
      font-size: 1.25rem;
    }
    @include m {
      font-size: 1.25rem;
    }
  }

  &__subtitle {
    align-self: center;
    margin: 1.25rem 0;
    text-align: center;
    color: $grey;
    @include s {
      font-size: 0.875rem;
    }
    @include m {
      font-size: 0.875rem;
    }
  }
}

.data {
  flex-direction: row;
  margin-bottom: 1.25rem;
  justify-content: center;
  align-items: center;


  @include s {
    flex: 1;
    flex-direction: column;
    justify-content: stretch;
    align-items: stretch;
  }
  @include m {
    flex: 1;
    flex-direction: column;
    justify-content: stretch;
    align-items: stretch;
  }
  
  &__field {
    flex-direction: row;
    background-color: $secondary-color;
    border-radius: 0.3125rem;
    justify-content: space-between;
    align-items: center;
    padding: 0.625rem;
    &:nth-of-type(2) {
      margin: 0 0.625rem;
    }
    @include s {
      flex: 1;
      flex-grow: 1;
      align-self: stretch;
      margin: 0.625rem 0;
      &:nth-of-type(2) {
        margin: 0.625rem 0;
      }
    }
    @include m {
      flex: 1;
      flex-grow: 1;
      align-self: stretch;
      margin: 0.625rem 0;
      &:nth-of-type(2) {
        margin: 0.625rem 0;
      }
    }
    @include l {
      margin: 0;
    }
  }


  &__input {
    font-size: 1rem;
    line-height: 1.3125rem;
    text-align: center;
    max-width: 2.5rem;
    border-radius: 0.3125rem;
    color: $secondary-dark;
    font-family: Montserrat;
    font-weight: 700;

    &:last-of-type {
      max-width: 3rem;
      max-height: 1.25rem;
    }
  }

  &__text {
    font-size: 1.25rem;
    font-weight: 700;
    margin-right: 0.3125rem;
    color: $white;
    @include s {
      font-size: 0.875rem;
    }
    @include m {
      font-size: 0.875rem;
    }
    @include l {
      font-size: 1rem;
    }
  }
}

.content {
  flex: 1;
  flex-direction: row;
  @include s {
    flex-direction: column;
  }
  @include m {
    flex-direction: column;
  }
}