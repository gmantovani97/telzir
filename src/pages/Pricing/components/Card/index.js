import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

import Icon from '@mdi/react';
import { mdiCheckBold, mdiClose, mdiSeal } from '@mdi/js';

import './styles.scss';

export default function Card({ best, data, choices }) {
  console.log(choices);
  console.log(data);
  const { maximumTime, tax, fee } = data;
  const { originValue, destinationValue, minutes } = choices;
  const exceededMinutes = maximumTime - minutes;
  const [price, setPrice] = useState({
    priceWithDiscount:
      exceededMinutes < 0 ? Math.abs(exceededMinutes) * (tax * fee) : 0,
    priceWithoutDiscount: maximumTime * tax,
  });

  useEffect(() => {
    setPrice({
      priceWithDiscount:
        exceededMinutes < 0 ? Math.abs(exceededMinutes) * (tax * fee) : 0,
      priceWithoutDiscount: mdiAccountMultiplePlusOutline * tax,
    });
    console.log(price);
  }, []);

  return (
    <div className={`card ${best && 'card--best'}`}>
      <img src={data.image} alt="Phone" className="image" />
      <h3 className="title">{data.name}</h3>
      {best && (
        <div className="flag">
          <Icon className="flag__icon" path={mdiSeal} />
          <p className="flag__text">Melhor plano</p>
        </div>
      )}
      <div className="section">
        <div className="section__item">
          <Icon
            className="section__icon section__icon--include"
            path={mdiCheckBold}
          />
          <p className="section__text">Ligações locais gratuitas</p>
        </div>
        <div className="section__item">
          <Icon
            className="section__icon section__icon--include"
            path={mdiCheckBold}
          />
          <p className="section__text">
            Ligações para mesma operadora sem custo
          </p>
        </div>
        <div className="section__item">
          <Icon
            className="section__icon section__icon--exclude"
            path={mdiClose}
          />
          <p className="section__text">
            Ligações entre 00:00 e 06:00 com 50% de desconto
          </p>
        </div>
        <div className="section__item">
          <Icon
            className="section__icon section__icon--exclude"
            path={mdiClose}
          />
          <p className="section__text">
            Ligações em feriado com 20% de desconto
          </p>
        </div>
      </div>
      <div className="price">
        <div className="price__section">
          <h3 className="price__number">82,50</h3>
          <h4 className="price__text">/mês</h4>
        </div>
      </div>
      <div className={`button ${best && 'button--best'}`}>
        <div className={`button__text ${best && 'button__text--best'}`}>
          Contratar
        </div>
      </div>
    </div>
  );
}

Card.propTypes = {
  best: PropTypes.bool,
  image: PropTypes.string.isRequired,
  data: PropTypes.shape.isRequired,
  choices: PropTypes.shape.isRequired,
};

Card.defaultProps = {
  best: false,
};
