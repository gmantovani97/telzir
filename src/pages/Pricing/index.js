import React, { useState, useMemo } from 'react';

import Rookie from '../../assets/images/rookie.svg';
import Prime from '../../assets/images/prime.svg';
import Elite from '../../assets/images/elite.svg';

import Card from './components/Card';
import './styles.scss';

export default function Pricing() {
  const prices = {
    1116: 1.9,
    1611: 2.9,
    1117: 1.7,
    1711: 2.7,
    1118: 0.9,
    1811: 1.9,
  };
  const [originValue, setOriginValue] = useState(18);
  const [destinationValue, setDestinationValue] = useState(11);
  const [minutes, setMinutes] = useState(200);
  const tax = useMemo(
    () => prices[Number(`${originValue}${destinationValue}`)],
    [originValue, destinationValue]
  );
  const values = useMemo(() => {
    const array = [
      {
        id: 1,
        name: 'FaleMais 30',
        image: Rookie,
        best: false,
        maximumTime: 30,
        priceWithDiscount:
          30 - minutes < 0
            ? (Math.abs(30 - minutes) * (tax * 1.1)).toFixed(2)
            : (0.0).toFixed(2),
        priceWithoutDiscount: (minutes * tax).toFixed(2),
        tax,
      },
      {
        id: 2,
        name: 'FaleMais 60',
        image: Prime,
        best: false,
        maximumTime: 60,
        priceWithDiscount:
          60 - minutes < 0
            ? (Math.abs(60 - minutes) * (tax * 1.1)).toFixed(2)
            : (0.0).toFixed(2),
        priceWithoutDiscount: (minutes * tax).toFixed(2),
        tax,
      },
      {
        id: 3,
        name: 'FaleMais 120',
        image: Elite,
        best: false,
        maximumTime: 120,
        priceWithDiscount:
          120 - minutes < 0
            ? (Math.abs(120 - minutes) * (tax * 1.1)).toFixed(2)
            : (0.0).toFixed(2),
        priceWithoutDiscount: (minutes * tax).toFixed(2),
        tax,
      },
    ];
    let maxValue = Infinity;
    let key;
    array.forEach((item, index) => {
      if (maxValue > item.priceWithDiscount) {
        maxValue = item.priceWithDiscount;
        key = index;
      }
    });
    array[key].best = true;
    return array;
  }, [minutes, tax]);

  return (
    <div className="container">
      <h1 className="container__title">
        Insira os dados abaixo para avaliar suas opções
      </h1>
      <h4 className="container__subtitle">
        Fale até 120 minutos sem pagar nada
      </h4>
      <div className="data">
        <div className="data__field">
          <p className="data__text">DDD de origem</p>
          <input
            maxLength={2}
            type="text"
            className="data__input"
            value={originValue}
            defaultValue={originValue}
            onChange={e => setOriginValue(e.target.value)}
          />
        </div>
        <div className="data__field">
          <p className="data__text">DDD de destino</p>
          <input
            data-testid="destination_input"
            maxLength={2}
            type="text"
            className="data__input"
            value={destinationValue}
            defaultValue={destinationValue}
            onChange={e => setDestinationValue(e.target.value)}
          />
        </div>
        <div className="data__field">
          <p className="data__text">Tempo da ligação (min)</p>
          <input
            data-testid="minutes_input"
            maxLength={3}
            type="text"
            className="data__input"
            value={minutes}
            defaultValue={minutes}
            onChange={e => setMinutes(e.target.value)}
          />
        </div>
      </div>
      <div className="content">
        {values.map(item => (
          <Card
            key={item.id}
            image={Rookie}
            data={item}
            choices={{ originValue, destinationValue, minutes }}
          />
        ))}
      </div>
    </div>
  );
}
