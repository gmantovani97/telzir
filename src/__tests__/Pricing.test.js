import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Pricing from '@/pages/Pricing';
import Card from '@/pages/Pricing/components/Card';

describe('Pricing component', () => {
  it('should calculate value of plans when minutes input change', () => {
    const { getByTestId, getByText } = render(<Pricing />);

    const input = getByTestId('minutes-input');
    const originSelect = getByTestId('origin-select');
    const destinationSelect = getByTestId('destination-select');

    fireEvent.change(originSelect, { target: { value: 18 } });
    fireEvent.change(destinationSelect, { target: { value: 11 } });
    fireEvent.change(input, { target: { value: '200' } });

    expect(Number(originSelect.value)).toBe(18);
    expect(Number(destinationSelect.value)).toBe(11);
    expect(input.value).toBe('200');

    expect(getByTestId('pricewithdiscount-3')).toContainElement(
      getByText('167.20')
    );
  });
});
