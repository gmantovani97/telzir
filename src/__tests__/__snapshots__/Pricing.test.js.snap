// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pricing component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h1
      class="container__title"
    >
      Insira os dados abaixo para avaliar suas opções
    </h1>
    <h4
      class="container__subtitle"
    >
      Fale até 120 minutos sem pagar nada
    </h4>
    <div
      class="data"
    >
      <div
        class="data__field"
      >
        <p
          class="data__text"
        >
          DDD de origem
        </p>
        <select
          data-testid="origin-select"
        >
          <option
            data-testid="origin-option-11"
            value="11"
          >
            11
          </option>
          <option
            data-testid="origin-option-16"
            value="16"
          >
            16
          </option>
          <option
            data-testid="origin-option-17"
            value="17"
          >
            17
          </option>
          <option
            data-testid="origin-option-18"
            value="18"
          >
            18
          </option>
        </select>
      </div>
      <div
        class="data__field"
      >
        <p
          class="data__text"
        >
          DDD de destino
        </p>
        <select
          data-testid="destination-select"
        >
          <option
            data-testid="destination-option-16"
            value="16"
          >
            16
          </option>
          <option
            data-testid="destination-option-17"
            value="17"
          >
            17
          </option>
          <option
            data-testid="destination-option-18"
            value="18"
          >
            18
          </option>
          <option
            data-testid="destination-option-11"
            disabled=""
            value="11"
          >
            11
          </option>
        </select>
      </div>
      <div
        class="data__field"
      >
        <p
          class="data__text"
        >
          Tempo da ligação (min)
        </p>
        <input
          class="data__input"
          data-testid="minutes-input"
          maxlength="3"
          type="text"
          value="20"
        />
      </div>
    </div>
    <div
      class="content"
    >
      <div
        class="card card--best"
      >
        <img
          alt="Phone"
          class="image"
          src="rookie.svg"
        />
        <h3
          class="title"
        >
          FaleMais 30
        </h3>
        <div
          class="flag"
        >
          <svg
            class="flag__icon"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M20.39,19.37L16.38,18L15,22L11.92,16L9,22L7.62,18L3.61,19.37L6.53,13.37C5.57,12.17 5,10.65 5,9A7,7 0 0,1 12,2A7,7 0 0,1 19,9C19,10.65 18.43,12.17 17.47,13.37L20.39,19.37M7,9L9.69,10.34L9.5,13.34L12,11.68L14.5,13.33L14.33,10.34L17,9L14.32,7.65L14.5,4.67L12,6.31L9.5,4.65L9.67,7.66L7,9Z"
            />
          </svg>
          <p
            class="flag__text"
          >
            Melhor plano
          </p>
        </div>
        <div
          class="section"
        >
          <div
            class="section__item"
          >
            <svg
              class="section__icon section__icon--include"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M9,20.42L2.79,14.21L5.62,11.38L9,14.77L18.88,4.88L21.71,7.71L9,20.42Z"
              />
            </svg>
            <p
              class="section__text"
            >
              Ligações locais gratuitas
            </p>
          </div>
          <div
            class="section__item"
          >
            <svg
              class="section__icon section__icon--include"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M9,20.42L2.79,14.21L5.62,11.38L9,14.77L18.88,4.88L21.71,7.71L9,20.42Z"
              />
            </svg>
            <p
              class="section__text"
            >
              Ligações para mesma operadora sem custo
            </p>
          </div>
          <div
            class="section__item"
          >
            <svg
              class="section__icon section__icon--exclude"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"
              />
            </svg>
            <p
              class="section__text"
            >
              Ligações entre 00:00 e 06:00 com 50% de desconto
            </p>
          </div>
          <div
            class="section__item"
          >
            <svg
              class="section__icon section__icon--exclude"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"
              />
            </svg>
            <p
              class="section__text"
            >
              Ligações em feriado com 20% de desconto
            </p>
          </div>
        </div>
        <div
          class="price"
        >
          <div
            class="price__section"
            data-testid="pricewithdiscount-1"
          >
            <h3
              class="price__number"
            >
              0.00
            </h3>
            <h4
              class="price__text"
            >
              /mês
            </h4>
          </div>
        </div>
        <div
          class="button button--best"
        >
          <p
            class="button__text button__text--best"
          >
            Contratar
          </p>
        </div>
      </div>
      <div
        class="card false"
      >
        <img
          alt="Phone"
          class="image"
          src="prime.svg"
        />
        <h3
          class="title"
        >
          FaleMais 60
        </h3>
        <div
          class="section"
        >
          <div
            class="section__item"
          >
            <svg
              class="section__icon section__icon--include"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M9,20.42L2.79,14.21L5.62,11.38L9,14.77L18.88,4.88L21.71,7.71L9,20.42Z"
              />
            </svg>
            <p
              class="section__text"
            >
              Ligações locais gratuitas
            </p>
          </div>
          <div
            class="section__item"
          >
            <svg
              class="section__icon section__icon--include"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M9,20.42L2.79,14.21L5.62,11.38L9,14.77L18.88,4.88L21.71,7.71L9,20.42Z"
              />
            </svg>
            <p
              class="section__text"
            >
              Ligações para mesma operadora sem custo
            </p>
          </div>
          <div
            class="section__item"
          >
            <svg
              class="section__icon section__icon--exclude"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"
              />
            </svg>
            <p
              class="section__text"
            >
              Ligações entre 00:00 e 06:00 com 50% de desconto
            </p>
          </div>
          <div
            class="section__item"
          >
            <svg
              class="section__icon section__icon--exclude"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"
              />
            </svg>
            <p
              class="section__text"
            >
              Ligações em feriado com 20% de desconto
            </p>
          </div>
        </div>
        <div
          class="price"
        >
          <div
            class="price__section"
            data-testid="pricewithdiscount-2"
          >
            <h3
              class="price__number"
            >
              0.00
            </h3>
            <h4
              class="price__text"
            >
              /mês
            </h4>
          </div>
        </div>
        <div
          class="button false"
        >
          <p
            class="button__text false"
          >
            Contratar
          </p>
        </div>
      </div>
      <div
        class="card false"
      >
        <img
          alt="Phone"
          class="image"
          src="elite.svg"
        />
        <h3
          class="title"
        >
          FaleMais 120
        </h3>
        <div
          class="section"
        >
          <div
            class="section__item"
          >
            <svg
              class="section__icon section__icon--include"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M9,20.42L2.79,14.21L5.62,11.38L9,14.77L18.88,4.88L21.71,7.71L9,20.42Z"
              />
            </svg>
            <p
              class="section__text"
            >
              Ligações locais gratuitas
            </p>
          </div>
          <div
            class="section__item"
          >
            <svg
              class="section__icon section__icon--include"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M9,20.42L2.79,14.21L5.62,11.38L9,14.77L18.88,4.88L21.71,7.71L9,20.42Z"
              />
            </svg>
            <p
              class="section__text"
            >
              Ligações para mesma operadora sem custo
            </p>
          </div>
          <div
            class="section__item"
          >
            <svg
              class="section__icon section__icon--exclude"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"
              />
            </svg>
            <p
              class="section__text"
            >
              Ligações entre 00:00 e 06:00 com 50% de desconto
            </p>
          </div>
          <div
            class="section__item"
          >
            <svg
              class="section__icon section__icon--exclude"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"
              />
            </svg>
            <p
              class="section__text"
            >
              Ligações em feriado com 20% de desconto
            </p>
          </div>
        </div>
        <div
          class="price"
        >
          <div
            class="price__section"
            data-testid="pricewithdiscount-3"
          >
            <h3
              class="price__number"
            >
              0.00
            </h3>
            <h4
              class="price__text"
            >
              /mês
            </h4>
          </div>
        </div>
        <div
          class="button false"
        >
          <p
            class="button__text false"
          >
            Contratar
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
